assets/wizards/popups/index.js

```diff
@@ -224,14 +198,64 @@ class PopupsWizard extends Component {
 		return `${ previewURL }?${ stringify( { ...options, newspack_popups_preview_id: id } ) }`;
 	};
 
+	updateAfterAPI = ( { campaigns, prompts, segments, settings } ) =>
+		this.setState( { campaigns, prompts, segments, settings } );
+
 	manageCampaignGroup = ( campaigns, method = 'POST' ) => {
 		const { setError, wizardApiFetch } = this.props;
 		return wizardApiFetch( {
 			path: '/newspack/v1/wizard/newspack-popups-wizard/batch-publish/',
 			data: { ids: campaigns.map( campaign => campaign.id ) },
 			method,
+			quiet: true,
+		} )
+			.then( this.updateAfterAPI )
+			.catch( error => setError( error ) );
+	};
+
+	renameCampaignGroup = ( id, name ) => {
+		const { setError, wizardApiFetch } = this.props;
+		return wizardApiFetch( {
+			path: `/newspack/v1/wizard/newspack-popups-wizard/rename-campaign/${ id }`,
+			method: 'POST',
+			data: { name },
+			quiet: true,
 		} )
-			.then( () => this.onWizardReady() )
+			.then( this.updateAfterAPI )
+			.catch( error => setError( error ) );
+	};
+
+	archiveCampaignGroup = ( id, status ) => {
+		const { setError, wizardApiFetch } = this.props;
+		return wizardApiFetch( {
+			path: `/newspack/v1/wizard/newspack-popups-wizard/archive-campaign/${ id }`,
+			method: status ? 'POST' : 'DELETE',
+			quiet: true,
+		} )
+			.then( this.updateAfterAPI )
+			.catch( error => setError( error ) );
+	};
+
+	duplicateCampaignGroup = ( id, name ) => {
+		const { setError, wizardApiFetch } = this.props;
+		return wizardApiFetch( {
+			path: `/newspack/v1/wizard/newspack-popups-wizard/duplicate-campaign/${ id }`,
+			method: 'POST',
+			data: { name },
+			quiet: true,
+		} )
+			.then( this.updateAfterAPI )
+			.catch( error => setError( error ) );
+	};
+
+	deleteCampaignGroup = id => {
+		const { setError, wizardApiFetch } = this.props;
+		return wizardApiFetch( {
+			path: `/newspack/v1/wizard/newspack-popups-wizard/delete-campaign/${ id }`,
+			method: 'DELETE',
+			quiet: true,
+		} )
+			.then( this.updateAfterAPI )
```

Comment:
After deletion, `history` should be updated - otherwise the UI is still displaying the deleted campaign's prompts, even though the top select is set to "Active prompts"

Reply:
This change is made in https://github.com/Automattic/newspack-plugin/pull/833/commits/2d6d1d1997ddcb214a3dcf7ddc2856baabb8bd52. Requires https://github.com/Automattic/newspack-popups/pull/429